### PR TITLE
Upgrade go1.19

### PR DESCRIPTION
**Pull request by maskarb**
* Upgrade dependencies and Go to 1.19
* Update deprecated calls to `ioutil` and `wait.PollImmediate`
* Update Makefile to align with Makefiles created by `OPERATOR_SDK_VERSION v1.32.0`

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: 1.18
+          go-version-file: go.mod
```
Comment (djnakabaale) [inline .github/workflows/ci.yaml:24]: nice  👍🏾 
no need to be updating go versions here after this change

File: main.go [hunk 2]
```diff
@@ -71,12 +75,11 @@ func main() {
...
-		MetricsBindAddress: metricsAddr,
-		Port:               9443,
-		LeaderElection:     enableLeaderElection,
-		LeaderElectionID:   "91c624a5.openshift.io",
-		Namespace:          watchNamespace,
+		Scheme:           scheme,
+		Metrics:          metricsserver.Options{BindAddress: metricsAddr},
+		LeaderElection:   enableLeaderElection,
+		LeaderElectionID: "91c624a5.openshift.io",
+		Cache:            cache.Options{DefaultNamespaces: map[string]cache.Config{watchNamespace: {}}},
```
Comment (djnakabaale) [inline main.go:82]: Just read about this and seems like caching on the ns could improve memory efficiency. Nice work  🎉

**Review by djnakabaale — DISMISSED** [review]
Thanks updating and cleaning up things here. Nice work 🎉

File: Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM registry.access.redhat.com/ubi8/go-toolset:1.18.9 as builder
+FROM registry.access.redhat.com/ubi8/go-toolset:1.19.10 AS builder
```
[the remark this answers is not in the document]
Comment (maskarb) [reply Dockerfile:2]: I added the `RUN yum -y update && yum clean all` which pulls in the latest go-toolset which contains the CVE fix:
https://access.redhat.com/errata/RHSA-2023:5721

**Review by djnakabaale — APPROVED** [review]

**Pull request merged**